### PR TITLE
Order v2

### PR DESCRIPTION
**Pull request by galofir**
* order.json is an order object. By default the assets will be downloaded directly to the azure blob "planet-blob". 
* To set a new order edit the order.json and run the order.py file without any agruments.

### CODE REVIEW

File: orders/order.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "order_test",   
+    "delivery": {
+    	"azure_blob_storage": {
+        	"sas_token": "sv=2017-11-09&ss=bfqt&srt=sco&sp=rwdlacup&se=2018-11-29T22:45:03Z&st=2018-11-02T14:45:03Z&spr=https&sig=6cjThqxnu501oKQvdqH0n3%2FlSa%2F30VXDWk%2FX3M9ui0g%3D",
```
Comment (taihak94) [inline orders/order.json]: maybe don't leave this here?

File: orders/orders.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    print("order created, processing...")
+    return response['_links']['_self']
+
+def download_order(items):
+    for item in items:
+        download_url = item['location']        
+        response = session.get(download_url)
+        print(response)  
+        
+# orders = session.get(order_url).json()
```
Comment (taihak94) [inline orders/orders.py]: remove comment

File: orders/server.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+from orders import download_order
+import json
+
+app = Flask(__name__)
+@app.route('/download', methods=['POST'])
+def download():
+    # check the order status
+    download_order(request.json['items'])
+    return 'downloaded'
+#app.run(debug=False, port=8080)
```
Comment (taihak94) [inline orders/server.py]: remove

File: orders/order.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      }  
+    },
+    "notifications": {
+      "email": false
+    },
+    "products": [
+    	{
+    		"item_type": "PSScene3Band",
+        	"item_ids": [
+            	"20181015_073916_1020"
```
Comment (taihak94) [inline orders/order.json]: but how do you know the ids? 
you should add a code the gets the ids based on a query for AOI\assets types\dates 
then activates the given items (they implemented bulk activation so it should be easier) and then gives the ids to the orders API

File: orders/order.json (hunk 1 above)
Comment (galofir) [reply orders/order.json]: I changed the sas token in azure so it's ok, but i'll delete it on the next PR

File: orders/orders.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+	else:
+		del order['delivery']['azure_blob_storage']
+	return order
+        
+def download_order(name, items):
+	item_index = 0
+	download_dir = abspath(join(dirname(__file__), 'order_' + name))
+	os.makedirs(download_dir)
+	for item in items:
+		item_index = item_index + 1
```
Comment (taihak94) [inline orders/orders.py]: `item_index += 1`

File: orders/orders.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+	os.makedirs(download_dir)
+	for item in items:
+		item_index = item_index + 1 
+		dest = abspath(join(download_dir, 'result_' + str(item_index) + '.zip'))
+		download_url = item['location']   
+		res = session.get(download_url)
+		with open(dest, "wb") as my_file:
+			my_file.write(res.content)     
+                
+def check_order_state(order_link):
```
Comment (taihak94) [inline orders/orders.py]: the name of this function is somewhat misleading  - it returns True/False but it also do stuff so idk if it should be called check...

File: orders/orders.py [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+		file.close()
+		return data
+
+def set_order(order):
+	response = session.post(order_url, json=order)
+	content = response.json()
+	if (response.status_code != 202):
+		for detail in content['field']['Details']:
+			print(detail['message'] + '\n')			
+		raise Exception('Error: ' + response.reason)
```
Comment (taihak94) [inline orders/orders.py]: you never catch this exception.

File: orders/order_template.json [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        	"account": "planetsa",
+        	"container": "planet-blob",
+        	"storage_endpoint_suffix": "",
+        	"path_prefix": ""
+        } 
+    },
+    "notifications": {
+        "email": false
+    },
+    "products": [
```
Comment (taihak94) [inline orders/order_template.json]: why do we need this?

File: orders/orders.py [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+	# in case the order is partialy failed	
+	elif (content['error_hints']):
+		for error in content['error_hints']:
+			print(error + '\n')	
+	print("order created, processing...")
+	return content['_links']['_self']
+
+def create_order(config, products):
```
Comment (taihak94) [inline orders/orders.py]: compose order

**Pull request merged**